pkg/rbd/controllerserver.go

```diff
@@ -240,6 +240,10 @@ func (cs *ControllerServer) checkSnapshot(req *csi.CreateVolumeRequest, rbdVol *
 		return status.Error(codes.NotFound, err.Error())
 	}
 
+	if (rbdVol.VolSize * util.MiB) < rbdSnap.SizeBytes {
```

Comment:
This check is not necessary in the CSI driver because it is already handled by the external-provisioner:
https://github.com/kubernetes-csi/external-provisioner/blob/master/pkg/controller/controller.go#L626

Reply:
@xing-yang Thanks for the review, the intention of adding this check is am not sure all the CO will do this validation or not, to safeguard the failure I have added this check

Reply:
@xing-yang is correct, it's not necessary in the Kubernetes context, the only thing I would mentoin however is that it is possible that at some point a different provisioner consumes the plugin.  The check is redundant and unnecessary, but at the same time it doesn't really hurt either.  Best case, it's never even hit, worst case we have a different CSI Provisioner some day and we miss this check somehow.  Not a huge deal, just an observation.